code/modules/spells/spell_types/rod_form.dm

```diff
@@ -11,48 +14,125 @@
 	invocation = "CLANG!"
 	invocation_type = INVOCATION_SHOUT
 	action_icon_state = "immrod"
+	/// The extra distance we travel per additional spell level.
+	var/distance_per_spell_rank = 3
+	/// The extra damage we deal per additional spell level.
+	var/damage_per_spell_rank = 20
 
-/obj/effect/proc_holder/spell/targeted/rod_form/cast(list/targets,mob/user = usr)
-	var/area/A = get_area(user)
-	if(istype(A, /area/wizard_station))
+/obj/effect/proc_holder/spell/targeted/rod_form/cast(list/targets, mob/user = usr)
+	var/area/our_area = get_area(user)
+	if(istype(our_area, /area/wizard_station))
 		to_chat(user, span_warning("You know better than to trash Wizard Federation property. Best wait until you leave to use [src]."))
 		return
-	for(var/mob/living/M in targets)
-		var/turf/start = get_turf(M)
-		var/obj/effect/immovablerod/wizard/W = new(start, get_ranged_target_turf(start, M.dir, (15 + spell_level * 3)))
-		W.wizard = M
-		W.max_distance += spell_level * 3 //You travel farther when you upgrade the spell
-		W.damage_bonus += spell_level * 20 //You do more damage when you upgrade the spell
-		W.start_turf = start
-		M.forceMove(W)
-		M.notransform = TRUE
-		M.status_flags |= GODMODE
-
-//Wizard Version of the Immovable Rod
 
+	// You travel farther when you upgrade the spell.
+	var/rod_max_distance = BASE_WIZ_ROD_RANGE + (spell_level * distance_per_spell_rank)
+	// You do more damage when you upgrade the spell.
+	var/rod_damage_bonus = (spell_level * damage_per_spell_rank)
+
+	for(var/mob/living/caster in targets)
+		var/turf/start_turf = get_turf(caster)
+		new /obj/effect/immovablerod/wizard(
+			start_turf,
+			start = start_turf,
+			end = get_ranged_target_turf(start_turf, caster.dir, (rod_max_distance + 2)), // Just a bit over the distance we got
+			wizard = caster,
+			max_distance = rod_max_distance,
+			damage_bonus = rod_damage_bonus,
+		)
+
+/// Wizard Version of the Immovable Rod.
 /obj/effect/immovablerod/wizard
-	var/max_distance = 13
-	var/damage_bonus = 0
-	var/turf/start_turf
 	notify = FALSE
 	dnd_style_level_up = FALSE
+	/// The wizard who's piloting our rod.
+	var/datum/weakref/our_wizard
+	/// The distance the rod will go.
+	var/max_distance = BASE_WIZ_ROD_RANGE
+	/// The damage bonus of the rod when it smacks people.
+	var/damage_bonus = 0
+	/// The turf the rod started from, to calcuate distance.
+	var/turf/start_turf
+
+/obj/effect/immovablerod/wizard/New(atom/start, atom/end, aimed_at, force_looping, mob/living/wizard, max_distance, damage_bonus)
```

Comment:
Looks like for some reason the parent is `New()`. Can you change both to use `Initialize()` instead?
The only significant difference is that the first argument becomes `mapload`.